baybe/utils/chemistry.py

```diff
@@ -72,158 +71,120 @@ def name_to_smiles(name: str) -> str:
 
 @lru_cache(maxsize=None)
 @_disk_cache
-def _smiles_to_mordred_features(smiles: str) -> np.ndarray:
-    """Memory- and disk-cached computation of Mordred descriptors.
+def _molecule_to_fingerprint_features(
+    fingerprint_encoder: BaseFingerprintTransformer,
+    molecule: str | Chem.PropertyMol.PropertyMol,
+) -> np.ndarray:
+    """Compute molecular fingerprint for a single SMILES string.
```

Comment:
The notion of "smiles" vs. "SMILES" is very inconsistent, we should decide on one variant and stick to it.